futures-util/src/stream/forward.rs

```diff
@@ -90,8 +84,9 @@ where
                 Poll::Ready(Some(Err(e))) => return Poll::Ready(Err(e)),
                 Poll::Ready(None) => {
                     try_ready!(self.as_mut().sink().as_pin_mut().expect(INVALID_POLL)
-                                   .poll_close(lw));
-                    return Poll::Ready(Ok(self.as_mut().sink().take().unwrap()))
+                                   .poll_flush(lw));
+                    let _ = self.as_mut().sink().as_pin_mut().take().unwrap();
```

Comment:
I think this will have to be `self.as_mut().sink().set(None)`, currently you're calling `take` on a temporary `Option<Pin<&mut Si>>` which won't cause the actual `Sink` held to be dropped.

Reply:
Ah, good catch! Thanks, I'll fix that right away.